### PR TITLE
Add Avaje Validator Module

### PR DESCRIPTION
**Pull request by SentryMan**
- Adds new Validator Module
- modifies the signature of `MvcValidator` (So that avaje-validator can modify its message per `Accept-Language`)

### CODE REVIEW

**Review by edgarespinawt — APPROVED** [review]
love it! @kliushnichenko want to look?

File: docs/asciidoc/modules/avaje-validator.adoc [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+Bean validation via https://avaje.io/validator/[Avaje Validator].
+
+=== Usage
+
+1) Add the dependency:
+
+[dependency, artifactId="jooby-avaje-validator"]
+.
+
```
Comment (kliushnichenko) [inline docs/asciidoc/modules/avaje-validator.adoc:11]: I'd also add the part with the annotation processor configuration

File: modules/jooby-avaje-validator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      <artifactId>jooby</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.jooby</groupId>
+      <artifactId>jooby-validation</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <!-- Hibernate Validator -->
```
Comment (kliushnichenko) [inline modules/jooby-avaje-validator/pom.xml:31]: Avaje

File: modules/jooby-avaje-validator/src/main/java/io/jooby/avaje/validator/AvajeValidatorModule.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+ * }</pre>
+ *
+ * <p>Supports validation of a single bean, list, array, or map.
+ *
+ * <p>The module also provides a built-in error handler that catches {@link
+ * ConstraintViolationException} and transforms it into a {@link
+ * io.jooby.validation.ValidationResult}
+ *
+ * @authors kliushnichenko, SentryMan
+ * @since 3.2.10
```
Comment (kliushnichenko) [inline modules/jooby-avaje-validator/src/main/java/io/jooby/avaje/validator/AvajeValidatorModule.java:50]: 3.3.1, and actually in HibernateModule I also missed that...

File: docs/asciidoc/modules/avaje-validator.adoc [hunk 4]
```diff
@@ -0,0 +1,274 @@
+== Avaje Validator
```
Comment (kliushnichenko) [inline docs/asciidoc/modules/avaje-validator.adoc:1]: we also need a reference to this file here https://github.com/jooby-project/jooby/blob/3.x/docs/asciidoc/modules/modules.adoc#validation

**Review by kliushnichenko — COMMENTED** [review]
nice, just a few minor comments

**Review by kliushnichenko — APPROVED** [review]

File: modules/jooby-avaje-inject/pom.xml [hunk 5]
```diff
@@ -11,7 +11,11 @@
 
   <modelVersion>4.0.0</modelVersion>
   <artifactId>jooby-avaje-inject</artifactId>
-
+  
+  <properties>
+    <maven.compiler.proc>full</maven.compiler.proc>
```
Comment (jknack) [inline modules/jooby-avaje-inject/pom.xml:16]: @SentryMan maven/javac compiler fails with `-proc:full` on JDK 17.

     Fatal error compiling: error: invalid flag: -proc:full 

Any idea of why?

Comment (SentryMan) [reply modules/jooby-avaje-inject/pom.xml:16]: What JDK 17 version do you have, they back ported this feature from 21 a while back

Comment (jknack) [reply modules/jooby-avaje-inject/pom.xml:16]: I see.

```
openjdk version "17.0.10" 2024-01-16
OpenJDK Runtime Environment Temurin-17.0.10+7 (build 17.0.10+7)
OpenJDK 64-Bit Server VM Temurin-17.0.10+7 (build 17.0.10+7, mixed mode)
```

Will update my sdk.

Comment (jknack) [reply modules/jooby-avaje-inject/pom.xml:16]: Fixed. Thanks.

**Pull request merged**